components/raftstore/src/store/worker/region.rs

```diff
@@ -356,22 +558,15 @@ where
     // we may delay some apply tasks if level 0 files to write stall threshold,
     // pending_applies records all delayed apply task, and will check again later
     pending_applies: VecDeque<Task<EK::Snapshot>>,
-    // Ranges that have been logically destroyed at a specific sequence number. We can
-    // assume there will be no reader (engine snapshot) newer than that sequence number. Therefore,
-    // they can be physically deleted with `DeleteFiles` when we're sure there is no older
-    // reader as well.
-    // To protect this assumption, before a new snapshot is applied, the overlapping pending ranges
-    // must first be removed.
-    // The sole purpose of maintaining this list is to optimize deletion with `DeleteFiles`
-    // whenever we can. Errors while processing them can be ignored.
-    pending_delete_ranges: PendingDeleteRanges,
 
     engine: EK,
     mgr: SnapManager,
     coprocessor_host: CoprocessorHost<EK>,
     router: R,
     pd_client: Option<Arc<T>>,
-    pool: FuturePool,
+    snap_gen_pool: FuturePool,
```

Comment:
Also is snap applying pool.

Reply:
Actually this pool is only used for snap generation; it's only used for spawning the `SnapGenContext::handle_gen` function.

The snap apply task runs on the region worker thread. The destroy task was also running on the region worker thread, but after this PR, it will run on a dedicated thread pool, similar to how we handle the snap gen task.